geodiff/src/geodiff.cpp

```diff
@@ -693,6 +693,111 @@ int GEODIFF_dumpData( const char *driverName, const char *driverExtraInfo, const
+        columnData.push_back( "\"type\": \"" + TableColumnType::baseTypeToString( column.type.baseType ) + "\"" );
+        columnData.push_back( "\"type_db\": \"" + jsonQuoted( column.type.dbType ) + "\"" );
+        if ( column.isPrimaryKey )
+          columnData.push_back( "\"primary_key\": true" );
+        if ( column.isNotNull )
+          columnData.push_back( "\"not_null\": true" );
+        if ( column.isAutoIncrement )
+          columnData.push_back( "\"auto_increment\": true" );
+        if ( column.isGeometry )
+        {
+          std::vector<std::string> geometryData;
+          geometryData.push_back( "\"type\": \"" + jsonQuoted( column.geomType ) + "\"" );
+          geometryData.push_back( "\"srs_id\": \"" + std::to_string( column.geomSrsId ) + "\"" );
+          if ( column.geomHasZ )
+            geometryData.push_back( "\"has_z\": true" );
+          if ( column.geomHasM )
+            geometryData.push_back( "\"has_m\": true" );
+
+          std::string geometryJson;
+          geometryJson += "\"geometry\": {\n               ";
+          geometryJson += join( geometryData.begin(), geometryData.end(), ",\n               " );
+          geometryJson += "\n            }";
+          columnData.push_back( geometryJson );
+        }
+
+        std::string columnJson;
+        columnJson += "         {\n";
+        columnJson += "            ";
+        columnJson += join( columnData.begin(), columnData.end(), ",\n            " );
+        columnJson += "\n         }";
+
+        columnsJson.push_back( columnJson );
+      }
+
+      std::vector<std::string> tableData;
+      tableData.push_back( "\"table\": \"" + jsonQuoted( tableName ) + "\"" );
+      tableData.push_back( "\"columns\": [\n" + join( columnsJson.begin(), columnsJson.end(), ",\n" ) + "\n         ]" );
+      if ( tbl.crs.srsId != 0 )
+      {
+        std::vector<std::string> crsData;
+        crsData.push_back( "\"srs_id\": " + std::to_string( tbl.crs.srsId ) );
+        crsData.push_back( "\"auth_name\": \"" + tbl.crs.authName + "\"" );
+        crsData.push_back( "\"auth_code\": " + std::to_string( tbl.crs.authCode ) );
+        crsData.push_back( "\"wkt\": \"" + jsonQuoted( tbl.crs.wkt ) + "\"" );
+
+        std::string crsJson;
+        crsJson += "\"crs\": {\n            ";
+        crsJson += join( crsData.begin(), crsData.end(), ",\n            " );
+        crsJson += "\n         }";
+        tableData.push_back( crsJson );
+      }
+      std::string tableJson;
+      tableJson += "      {\n         ";
+      tableJson += join( tableData.begin(), tableData.end(), ",\n         " );
+      tableJson += "\n      }";
+
+      tablesData.push_back( tableJson );
+    }
+    std::string res = "{\n   \"geodiff_schema\": [\n";
+    res += join( tablesData.begin(), tablesData.end(), ",\n   " );
+    res += "\n   ]\n";
+    res += "}\n";
+
+    // write file content
+    flushString( json, res );
```

Comment:
would be nice to have also command line option `--human` vs `machine` JSON. I assume we can have JSON without whitespaces and linebreaks on mergin server to save a bit of space on disk?

Reply:
To be done in #125